### ISSUE TITLE
Why is hazelcast-kubernetes trying to access k8s API in DNS lookup mode?

### ISSUE THREAD

**Issue by sklakegg** [hazelcast/hazelcast-kubernetes#326]
I have the following configuration for using hazelcast with vertx:
```
JoinConfig joinConfig = new JoinConfig();
joinConfig.getMulticastConfig().setEnabled(false);
joinConfig.getTcpIpConfig().setEnabled(false);
joinConfig.getKubernetesConfig().setEnabled(true);

Map<String, Comparable> discoveryProperties = new HashMap<>();
discoveryProperties.put("service-dns", System.getProperty("HZ_SERVICE"));

List<DiscoveryStrategyConfig> discoveryStrategyConfigs = new ArrayList<>();
discoveryStrategyConfigs.add(new DiscoveryStrategyConfig()
        .setDiscoveryStrategyFactory(new HazelcastKubernetesDiscoveryStrategyFactory())
        .setClassName("com.hazelcast.kubernetes.HazelcastKubernetesDiscoveryStrategy")
        .setProperties(discoveryProperties));
joinConfig.getDiscoveryConfig().setDiscoveryStrategyConfigs(discoveryStrategyConfigs);

Config hazelcastConfig = ConfigUtil.loadConfig();
hazelcastConfig.getNetworkConfig().setJoin(joinConfig);
HazelcastClusterManager hazelcastClusterManager = new HazelcastClusterManager();
hazelcastClusterManager.setConfig(hazelcastConfig);

vOptions.setClusterManager(hazelcastClusterManager);
```

Everything works and a memberships between the two pods is formed without issues. However, it seems that hazelcast-kubernetes tries to call the kubernets API since I get the following:

```
java.io.IOException: Server returned HTTP response code: 403 for URL: https://kubernetes.default.svc/api/v1/namespaces/default/pods
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1932)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1528)
	at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:527)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:308)
	at com.hazelcast.kubernetes.RestClient.checkHttpOk(RestClient.java:131)
	at com.hazelcast.kubernetes.RestClient.call(RestClient.java:111)
	at com.hazelcast.kubernetes.RestClient.get(RestClient.java:79)
	at com.hazelcast.kubernetes.KubernetesClient$1.call(KubernetesClient.java:503)
	at com.hazelcast.kubernetes.KubernetesClient$1.call(KubernetesClient.java:497)
	at com.hazelcast.kubernetes.RetryUtils.retry(RetryUtils.java:53)
	at com.hazelcast.kubernetes.KubernetesClient.callGet(KubernetesClient.java:497)
	at com.hazelcast.kubernetes.KubernetesClient.endpoints(KubernetesClient.java:80)
	at com.hazelcast.kubernetes.KubernetesApiEndpointResolver.resolve(KubernetesApiEndpointResolver.java:70)
	at com.hazelcast.kubernetes.HazelcastKubernetesDiscoveryStrategy.discoverNodes(HazelcastKubernetesDiscoveryStrategy.java:135)
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.discoverNodes(DefaultDiscoveryService.java:71)
	at com.hazelcast.internal.cluster.impl.DiscoveryJoiner.getPossibleAddresses(DiscoveryJoiner.java:70)
	at com.hazelcast.cluster.impl.TcpIpJoiner.searchForOtherClusters(TcpIpJoiner.java:426)
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.searchForOtherClusters(SplitBrainHandler.java:75)
	at com.hazelcast.internal.cluster.impl.SplitBrainHandler.run(SplitBrainHandler.java:42)
	at com.hazelcast.spi.impl.executionservice.impl.DelegateAndSkipOnConcurrentExecutionDecorator$DelegateDecorator.run(DelegateAndSkipOnConcurrentExecutionDecorator.java:77)
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
```

I thought I could avoid providing access to the API for the hazelcast service when I used the headless dns mode?

**Comment by leszko** [comment]
When you use the [DNS Lookup mode ](https://github.com/hazelcast/hazelcast-kubernetes#dns-lookup), then Hazelcast should not try to access Kubernetes API. Which hazelcast-kubernetes plugin version do you use? Do you encounter the same issue with the latest one?

**Comment by sklakegg** [comment]
I use v1.5.5 from Dec 2020:

<dependency>
    <groupId>com.hazelcast</groupId>
    <artifactId>hazelcast-kubernetes</artifactId>
    <version>1.5.5</version>
</dependency>

I have not upgraded to 2.2.2 as I belive I would have to update to vert.x v4 as well if I'm not mistaken. Would that be the only fix?

**Comment by leszko** [comment]
I think v1.5.5 should work fine. It seems that your Hazelcast instance somehow does not use the configuration you specified. Could you some minimal reproducer? Or basically try to reproduce it without Vertx?

**Comment by sklakegg** [comment]
Hi again, yes I will try once I get some spare time. Quite hectic at the moment

**Comment by sklakegg** [comment]
I upgraded to the latest version of the plugin and it worked. It seems like the old version forced me to set an DiscoveryStrategyConfig, which in turned made it use the API. When I upgraded the plugin I could remove the part where I set the DiscoveryStrategyConfig, and it used the dns strategy without issues. At least that's what I can recall doing before summer, I did not spend much time on this.

Anyways, closing since this is no longer an issue in the newer version.
